src/lib/pgp-key.h

```diff
@@ -343,6 +348,34 @@ struct pgp_key_t {
      */
     pgp_subsig_t *latest_binding(bool validated = true);
 
+    /** @brief Returns true if signature is produced by the key itself. */
+    bool is_signer(const pgp_subsig_t &sig) const;
+
+    /** @brief Returns true if key is expired according to sig. */
+    bool is_expired(const pgp_subsig_t &sig) const;
+
+    /** @brief Check whether signature is key's self certification. */
+    bool is_self_cert(const pgp_subsig_t &sig) const;
+
+    /** @brief Check whether signature is key's direct-key self-signature */
+    bool is_direct_self(const pgp_subsig_t &sig) const;
+
+    /** @brief Check whether signature is key's/subkey's revocation */
+    bool is_revocation(const pgp_subsig_t &sig) const;
+
+    /** @brief Check whether signature is userid revocation */
+    bool is_uid_revocation(const pgp_subsig_t &sig) const;
+
+    /** @brief Check whether signature is subkey binding */
+    bool is_binding(const pgp_subsig_t &sig) const;
+
+    /**
+     * @brief Validate key-related signature, where this is a signing key.
```

Comment:
It is not clear what does this sentence mean.
I guess it is a replacement for `pgp_key_validate_signature()` which was documented this way:
> Validate key's signature.

Reply:
Thanks, will update the description.